spec.bs

```diff
@@ -2663,11 +2660,18 @@ To <dfn>report result</dfn> given a [=leading bid info=] |leadingBidInfo|, a
   1. Let |igAd| be the [=interest group ad=] from |winner|'s [=generated bid/interest group=]'s
     [=interest group/ads=] whose [=interest group ad/render url=] is |winner|'s
     [=generated bid/ad descriptor=]'s [=ad descriptor/url=].
-  1. If |igAd|'s [=interest group ad/buyer and seller reporting ID=]
-     [=map/exists=] and the result of [=query reporting ID k-anonymity count=]
-     given |winner|'s [=generated bid/interest group=] and |igAd| is true, then [=map/set=]
-     |browserSignals|["{{ReportingBrowserSignals/buyerAndSellerReportingId}}"] to |igAd|'s
-     [=interest group ad/buyer and seller reporting ID=].
+  1. If the |winner|'s [=generated bid/selected buyer and seller reporting ID=] is not null and the
+    result of running [=query reporting ID k-anonymity count=] with |winner|'s

```

Comment:
I'm not very familiar with the k-anon work, so just want to double check that this matches our implementation: The reporting ids need to use this algorithm to query the k-anon server, but cannot be queried from the k-anon cache, right?
It seems the [=update k-anonymity cache for interest group=] does update a hash with "selected buyer and seller reporting ID", but not other reporting ids. So whenever those reporting ids need to be checked against k-anon, they need to check from the k-anon server?

Reply:
Good point, Qingxin. This case, like the others here, does indeed use the k-anon cache, so it should presumably be something more like ...

```
1. If the |winner|'s [=generated bid/selected buyer and seller reporting ID=] is not null:
   1. Let |reportingHashCode| be the result of [=compute the key hash of reporting ID=] with |winner|'s
      [=generated bid/interest group=], |igAd|, and |winner|'s
      [=generated bid/selected buyer and seller reporting ID=].
   1. If [=query k-anonymity cache=] for |reportingHashCode| returns true:
      1. [=map/Set=] |browserSignals|["{{ReportingBrowserSignals/selectedBuyerAndSellerReportingId}}"] ...
```

Reply:
SG. Once it's changed, it'll be ready to go.

Reply:
Thank you guys, i just made the change so let me know if the wording sounds good!

Reply:
it seems [[=query reporting ID k-anonymity count=]](https://pr-preview.s3.amazonaws.com/HabibiYou/turtledove/pull/1275.html#query-reporting-id-k-anonymity-count) itself should query the cache, instead of "[querying the k-anonymity count](https://pr-preview.s3.amazonaws.com/HabibiYou/turtledove/pull/1275.html#query-k-anonymity-count)", like what [[=query generated bid k-anonymity count=]](https://pr-preview.s3.amazonaws.com/HabibiYou/turtledove/pull/1275.html#query-generated-bid-k-anonymity-count) does. Then these call sites can keep what you had before (i.e., call [=query reporting ID k-anonymity count=] directly).
Let me confirm with Russ since they added the cache part.